pkg/roachprod/roachprod.go

```diff
@@ -1963,6 +1963,110 @@ func isWorkloadCollectorVolume(v vm.Volume) bool {
 	return false
 }
 
+const (
+	otelCollectorPort   = 4317
+	jaegerUIPort        = 16686
+	jaegerContainerName = "jaeger"
+	jaegerImageName     = "jaegertracing/all-in-one:latest"
+)
+
+// StartJaeger starts a jaeger instance on the last node in the given
+// cluster and configures the cluster to use it.
+func StartJaeger(
+	ctx context.Context, l *logger.Logger, clusterName string, virtualClusterName string, secure bool,
+) error {
+	if err := LoadClusters(); err != nil {
+		return err
+	}
+	c, err := newCluster(l, clusterName, install.SecureOption(secure))
+	if err != nil {
+		return err
+	}
+
+	// TODO(ssd): Currently this just uses the all-in-one docker
+	// container with in memory storage. Might be nicer to just
+	// install from source or get linux binaries and start them
+	// with systemd. For now this just matches what we've been
+	// copy and pasting.
+	jaegerNode := c.TargetNodes()[len(c.TargetNodes())-1:]
+	err = install.InstallTool(ctx, l, c, jaegerNode, "docker", l.Stdout, l.Stderr)
+	if err != nil {
+		return err
+	}
+	startCmd := fmt.Sprintf("docker run -d --name %s -p %[2]d:%[2]d -p %[3]d:%[3]d %s",
+		jaegerContainerName,
+		otelCollectorPort,
+		jaegerUIPort,
+		jaegerImageName)
+	err = c.Run(ctx, l, l.Stdout, l.Stderr, jaegerNode, "start jaegertracing/all-in-one using docker", startCmd)
+	if err != nil {
+		return err
+	}
+
+	otelCollectionHost, err := c.GetInternalIP(jaegerNode[0])
+	if err != nil {
+		return err
+	}
+
+	otelCollectionHostPort := net.JoinHostPort(otelCollectionHost, strconv.Itoa(otelCollectorPort))
+	_, err = c.ExecSQL(ctx, l, jaegerNode, virtualClusterName, 0, []string{"-e", fmt.Sprintf(
+		"SET CLUSTER SETTING trace.opentelemetry.collector='%s'", otelCollectionHostPort)})
```

Comment:
Should this be its own cmd, or the error non-fatal, so you can install the tool before/without sql running on the same cluster and indeed node?

Reply:
I've made this opt-in via an option and now the command prints the config to run.

I'm not opposed to making it another command just to run this option, but the odd thing about this, the grafana command, and the jaeger command is they assume things about what node is being used for what so there is inter-command state that you kinda just have to know to make things work out.